File: blog-hook/beautify.py
import sys
from BeautifulSoup import BeautifulSoup as bs

filename = sys.argv[1]

html_file = open(filename, 'r')
html = html_file.read()
html_file.close()

html_prettified = bs(html).prettify()

html_file = open(filename, 'w')
html_file.write(html_prettified)
html_file.close()

File: blog-hook/beautify.rb
# Remove the unexpected newline in html code between two Chinese characters
# TODO Current implementation doesn't take <pre> tag into consideration.
#      However, writing Chinese inside a <pre> tag is quite a rare case.

Dir["#{ARGV[0]}/**.html"].each do |html|
  text = File.read(html).gsub(/(?<=[^\p{ASCII}])\n *(?=[^a-zA-Z])/, '')
  File.write(html, text)
end
